Composer/packages/lib/indexers/src/dialogUtils/jsonDiff.ts

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import has from 'lodash/has';
+import get from 'lodash/get';
+import set from 'lodash/set';
+import isEqual from 'lodash/isEqual';
+import cloneDeep from 'lodash/cloneDeep';
+
+import { JsonWalk, VisitorFunc } from '../utils/jsonWalk';
+
+import { ListDiff } from './listDiff';
+
+export interface IJSONChangeAdd {
+  path: string;
+  value: any;
+}
+export interface IJSONChangeDelete {
+  path: string;
+  value: any;
+}
+export interface IJSONChangeUpdate {
+  path: string;
+  value: any;
+  preValue: any;
+}
+
+export interface IJsonChanges {
+  adds: IJSONChangeAdd[];
+  deletes: IJSONChangeDelete[];
+  updates: IJSONChangeUpdate[];
+}
+
+export type IComparator = (json1: any, json2: any, path: string) => { isChange: boolean; isStop: boolean };
+export type IStopper = (json1: any, json2: any, path: string) => boolean;
+
+export const JsonPathStart = '$';
+
+export function jsonPathToObjectPath(path: string) {
+  // eslint-disable-next-line security/detect-non-literal-regexp
+  const reg = new RegExp(`^\\${JsonPathStart}\\.`);
+  return path.replace(reg, '');
+}
+export function getWithJsonPath(value, path) {
+  if (path === JsonPathStart) return value;
+  const objPath = jsonPathToObjectPath(path);
+  return get(value, objPath);
+}
+export function hasWithJsonPath(value, path) {
+  if (path === JsonPathStart) return true;
+  const objPath = jsonPathToObjectPath(path);
+  return has(value, objPath);
+}
```

Comment:
Those 3 jPath related functions, what's the difference with standard jpath library provided and we must provide at here?